cdisc_rules_engine/utilities/sql_rule_processor.py

```diff
@@ -52,29 +53,40 @@ def __init__(
         self.cache = CacheServiceFactory(default_config).get_cache_service()
         self.library_metadata = library_metadata
 
-    # @classmethod
-    # def rule_applies_to_domain(cls, dataset_metadata: SDTMDatasetMetadata, rule: dict) -> bool:
-    #     """
-    #     Check that rule is applicable to dataset domain
-    #     """
-    #     domains = rule.get("domains") or {}
-    #     include_split_datasets: bool = domains.get("include_split_datasets")
-
-    #     included_domains = domains.get("Include", [])
-    #     excluded_domains = domains.get("Exclude", [])
-
-    #     is_included = cls._is_domain_name_included(dataset_metadata, included_domains, include_split_datasets)
-    #     is_excluded = cls._is_domain_name_excluded(dataset_metadata, excluded_domains)
-
-    #     # additional check for split domains based on the flag
-    #     is_excluded, is_included = cls._handle_split_domains(
-    #         dataset_metadata.is_split,
-    #         include_split_datasets,
-    #         is_excluded,
-    #         is_included,
-    #     )
-
-    #     return is_included and not is_excluded
+    @classmethod
+    def rule_applies_to_domain(cls, dataset_metadata: SQLDatasetMetadata, rule: dict) -> bool:
+        """Check that rule is applicable to dataset domain"""
+        domains = rule.get("domains") or {}
+
+        included_domains = domains.get("Include", [])
+        excluded_domains = domains.get("Exclude", [])
+
+        domain_name = dataset_metadata.dataset_name.upper()
+
+        is_supp = domain_name.startswith("SUPP")
+
+        if included_domains:
+            if ALL_KEYWORD in included_domains:
+                return True
+            if domain_name not in included_domains:
+                if not any(
+                    (pattern == "SUPP--" and is_supp) or (pattern == "AP--" and domain_name.startswith("AP"))
+                    for pattern in included_domains
+                ):
+                    return False
+
+        if excluded_domains:
```

Comment:
Are these two `if` statements supposed to be negations of one another? Is it worth extracting the logic and just returning the negation?